test/integration/dredd-test.coffee

```diff
@@ -608,3 +608,28 @@ describe 'Dredd class Integration', ->
       assert.notEqual(matches[2][1], 'skip')
     )
   )
+
+  describe('when using Swagger document with hooks', ->
+    reTransactionName = /hook: (.+)/g
+    matches = undefined
+
+    beforeEach((done) ->
+      execCommand(
+        options:
+          path: './test/fixtures/multiple-responses.yaml'
+          hookfiles: './test/fixtures/swagger-transaction-names.js'
+      , (err) ->
+        matches = []
```

Comment:
Has to be `matches` really filled two times? From my reading, it would be enough to do `matches = []` on line 614.